doc/reference/src/low-level-execution-model.pdc

```diff
@@ -111,276 +111,542 @@ The oracle-free abstract algorithm for optimal reduction operates on four node t
+  ~
+  \begin{subfigure}[c]{0.1\textwidth}
+    \begin{tikzpicture}
+      \inode[]{a}{$⊗$}
+      \inode[right = 20 pt]{b}{$⊗$}
+      \inwire[]{a}{a}
+      \inwire[]{b}{d}
+    \end{tikzpicture}
+  \end{subfigure}
 \end{figure}
 
 \begin{figure}[H]
-\caption{Eraser-eraser annihilation}
-\begin{subfigure}[c]{0.1\textwidth}
-\begin{tikzpicture}
-\inode[]{a}{$⊗$}
-\inode[below = 40 pt]{b}{$⊗$}
-\vwire{a}{b}
-\end{tikzpicture}
-\end{subfigure}
-~
-\begin{subfigure}[c]{0.05\textwidth}
-\huge{→}
-\end{subfigure}
-~
-\begin{subfigure}[c]{0.1\textwidth}
-\end{subfigure}
+  \caption{Eraser-eraser annihilation}
+  \begin{subfigure}[c]{0.1\textwidth}
+    \begin{tikzpicture}
+      \inode[]{a}{$⊗$}
+      \inode[below = 40 pt]{b}{$⊗$}
+      \vwire{a}{b}
+    \end{tikzpicture}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.05\textwidth}
+    \huge{→}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.1\textwidth}
+  \end{subfigure}
 \end{figure}
 
 \floatstyle{boxed}
 \restylefloat{figure}
 
 ### Argument for correctness
 
+
 ## Bespoke term encoding
 
 - Primitive types (integer, string, bytes) ~> node types w/data
 - n-arity primitive functions (+, -, *, concat, index) ~> rewrite rules
-- alt-term encoding: $lam x. x + x + x$ ~> $3 * x$ rewrite rule instead of multi-node graph
+- alt-term encoding: $\lambda x. x + x + x$ ~> $3 * x$ rewrite rule instead of multi-node graph
 
 - Over set of underlying opcodes & von-Neumann-ish VM (e.g. x86, LLVM, EVM).
 - Use linear types (e.g. rewrite memory locations in place)
 - Must be semantically equivalent to interaction system encoding after read-back
 - Must preserve any-order reduction (exists some maximum `n` rewrite steps after which graph read-back must have final form regardless of order)
 - Can relax diamond property (strong confluence)? unclear if necessary
 
+### Argument for Correct read-back
+
+We wish to show that a read-back of an EAL-expression that has undergone zero or more reductions gives back the same result as normal evaluation up to alpha equality.
+
+Before we can prove this result, we must first prove a few lemmas and theorems first as well talk about what fan in's with the same label imply.
+
+The first lemma we wish to prove is the following
+
+\begin{Lemma}{AST→Net has one free port}{AST->Net}
+  Let $A$ = be a valid BOHM ADT.
+  \\ \\
+  Now Consider the net over $A$, let $N_a$ be this net.
+  \\ \\
+  Now let us consider any sub case of $A$, $L = P(n_1 \star \cdots \star n_k)$.
+  \\ \\
+  The node corresponding to $L$ looks.
+
+  \begin{tikzpicture}
+    \draw (0,0) circle (.8cm);
+  \end{tikzpicture}
+
+  Where the labels in $N_a$ are the same as in $L$, with $n_{k+1}$ being an extra port which connects to the ADT above it.
+  \\ \\
+  For an ADT to be realized, we must have all arguments, so all these ports must be connected.
+  Now, if $L$ is $A$, then we have one free port, however if it is not, then this free port must not be free but instead be connected to the ADT above it.
```

Comment:
What does "above" mean?

I think you're getting at "must be connected to the encoding of an outer term".

I agree with this but I do not think it is trivial, we should argue each case of the encode function.

Reply:
above means, the ADT that contains L, if L is not the root expression.

> I agree with this but I do not think it is trivial, we should argue each case of the encode function.

yeah this addition wouldn't be hard, it necessitates that we give an algorithm for encode. As it is, this proof  works under nodes giving back the prots to connect to. This is alluded to in the proof.

Reply:
Ok this can be done later